ts: record child metrics with low frequency poller

jasonlmfong opened the pull request:
This change adds the ability to get changefeed child metrics to be recorded. The child metrics have their names augmented with their labels.

Epic: CRDB-55079
Release: None

angles-n-daemons commented on pkg/util/metric/metric.go:
```diff
@@ -230,11 +230,20 @@ func (m *Metadata) GetLabels(useStaticLabels bool) []*prometheusgo.LabelPair {
+	}
+	return *m.EnableLowFreqChildCollection
+}
+
```
This is an interesting thought. I'd actually prefer this to be a difficult operation, at least to start. That being said would it be difficult to change the default behavior in the future?

jasonlmfong replied:
The reason I thought this could be defaulted to true is because we have the cluster setting gating this feature already
but yeah I understand making it harder to start
in any case the default should be easy to change in the future

angles-n-daemons reviewed (commented):
Nice man! This looks good so far, added a couple of comments - but nothing really concerning to start.

jasonlmfong commented on pkg/server/status/recorder.go:
```diff
@@ -906,6 +943,93 @@ func (rr registryRecorder) recordChild(
+	labels := rr.registry.GetLabels()
+	rr.registry.Each(func(name string, v interface{}) {
+		// Filter for changefeed metrics only
+		if !strings.HasPrefix(name, "changefeed.") {
```
I see

jasonlmfong commented on pkg/server/status/recorder.go:
that makes sense

angles-n-daemons reviewed (approved):
nice, thanks again for doing all this!

jasonlmfong commented on pkg/util/metric/registry.go:
```diff
@@ -288,11 +289,49 @@ func ExportedName(name string) string {
+// EncodeLabeledName formats a metric name with labels in Prometheus format.
+// The labels are sanitized, sorted by key, and encoded as key="value" pairs
+// within curly braces. Example: metric_name{label1="value1",label2="value2"}
+func EncodeLabeledName(format string, name string, labels []*prometheusgo.LabelPair) string {
```
the main difference here is that prometheus standard would scrub the `values` while we don't NEED to, what are your thoughts on that?

jasonlmfong committed:
metrics: add CHANGEFEEDS category to changefeed metrics

These changefeed metrics had their categories as UNSET previously, this properly sets their category for usage.

Epic: None
Release: None

jasonlmfong committed:
ts: record child metrics with low frequency poller

This change adds the ability to get changefeed child metrics to be recorded. The child metrics have their names augmented with their labels. Each agg metric has up to 1024 child labels collected.

Epic: CRDB-55079
Release: None
